### DISCUSSION TITLE
Meson detects wrong compiler version

### DISCUSSION THREAD

**Discussion by bfamzz** [msys2/MINGW-packages#13091] [General]
Hello all,

I noticed when building with meson that it outputs the wrong gcc version as shown below.

<img width="506" alt="image" src="https://user-images.githubusercontent.com/38426051/189981140-3481c503-c7ad-4c20-8c4f-f3a26b0307da.png">

as compared to the output of this `gcc -v` shown below

<img width="952" alt="image" src="https://user-images.githubusercontent.com/38426051/189981338-60ea7bb8-a0f1-48aa-9570-cf64c4fab4d5.png">

Any idea why this is happening?

Thanks.

**Comment by Biswa96** [comment]
It is hard to find the reason behind the issue with this very short information. Some suggestions. Make sure there is no other msys2 or cygwin installation in PATH variable. Check if all packages are updated.

**Reply by bfamzz** [reply]
Thanks @Biswa96 There are no other msys2 or cygwin installations.

I think I found it. It's from the pacman cache as shown below.

<img width="680" alt="image" src="https://user-images.githubusercontent.com/38426051/190003857-bd2e759c-39a2-4e34-9c33-d5a8d77e5b9f.png">

Any idea on how to clean up the cache please?

Thanks.

**Reply by bfamzz** [reply]
> Thanks @Biswa96 There are no other msys2 or cygwin installations.
> 
> I think I found it. It's from the pacman cache as shown below.
> 
> <img alt="image" width="680" src="https://user-images.githubusercontent.com/38426051/190003857-bd2e759c-39a2-4e34-9c33-d5a8d77e5b9f.png">
> 
> Any idea on how to clean up the cache please?
> 
> Thanks.

Cleaning up the cache did not work alone. However, also creating a new build directory captured the correct gcc version as shown below.

<img width="568" alt="image" src="https://user-images.githubusercontent.com/38426051/190012968-e5751f91-39f9-453d-9e08-35558394edca.png">

**Reply by Biswa96** [reply]
Pacman cache has nothing to do with it. But the last screenshot shows 12.2.0 version in the output, isn't that right?

**Reply by bfamzz** [reply]
> Pacman cache has nothing to do with it. But the last screenshot shows 12.2.0 version in the output, isn't that right?

Yes it did. I wasn't too convinced it had anything to do with it either but I just shared what I did that supposedly worked.

Thanks for your help.
